hudi-spark-datasource/hudi-spark/src/main/java/org/apache/hudi/HoodieDataSourceHelpers.java

```diff
@@ -98,6 +106,20 @@ public static String latestCommit(HoodieStorage storage, String basePath) {
     return timeline.lastInstant().get().requestedTime();
   }
 
+  /**
+   * Returns the last successful write operation's completed instant.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  public static HoodieInstant latestCompletedCommitCompletionTime(FileSystem fs, String basePath) {
```

Comment:
Similar here on extracting the common functionality with `latestCommit` (two methods above).